src/test/java/com/amazon/titan/testutils/TravisCiHeartbeat.java

```diff
@@ -0,0 +1,126 @@
+ *
+ * The heartbeat is started and stopped before and after each test case. The current running unit test and the current
+ * execution time of the test is included in the heartbeat for better visibility in to any long running tests.
+ *
+ * @author Johan Jacobs
+ */
+public class TravisCiHeartbeat {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TravisCiHeartbeat.class);
+
+    private static final long DEFAULT_HEARTBEAT_INTERVAL    = 300000;
+    private static final int DEFAULT_MAXIMUM_INTERVALS      = 8;
+
+    private Timer heartbeatTimer;
+    private final ReentrantLock lock;
+    private boolean timerStarted;
+    private int intervals;
+    private long configuredHeartbeatInterval;
+    private int configuredMaximumIntervals;
+    private String configuredUnitTestName;
+    private long heartbeatStartTime;
+
+    public TravisCiHeartbeat() {
+        this(DEFAULT_HEARTBEAT_INTERVAL, DEFAULT_MAXIMUM_INTERVALS);
+    }
+
+    public TravisCiHeartbeat(long heartbeatInterval, int maximumIntervals) {
+
+        this.configuredHeartbeatInterval = heartbeatInterval;
+        this.configuredMaximumIntervals = maximumIntervals;
+        this.heartbeatStartTime = System.currentTimeMillis();
+
+        this.lock = new ReentrantLock();
+    }
+
+    public void startHeartbeat(String unitTestName) {
+
+        this.lock.lock();
+        try {
+            if(this.timerStarted) {
+                LOG.warn(String.format("Travis CI heartbeat timer is already running for unit test with name: %s.", this.configuredUnitTestName));
+                return;
+            }
+
+            this.configuredUnitTestName = unitTestName;
+            this.heartbeatTimer = new Timer("Unit test heartbeat timer");
+            this.heartbeatTimer.schedule(new TimerTask() {
+                @Override
+                public void run() {
+
+                    intervals++;
+                    long currentRunTimeMilliseconds = System.currentTimeMillis() - heartbeatStartTime;
+
+                    if (intervals == configuredMaximumIntervals) {
+                        LOG.warn(String.format("Heartbeat - [%d] - %s - %dms.",
+                                intervals,
+                                configuredUnitTestName,
+                                currentRunTimeMilliseconds));
+                        stopHeartbeat();
+                        return;
+                    }
+
+                    LOG.info(String.format("Heartbeat - [%d] - %s - %dms",
+                            intervals,
+                            configuredUnitTestName,
+                            currentRunTimeMilliseconds));
+                }
+            }, this.configuredHeartbeatInterval, this.configuredHeartbeatInterval);
+
+            this.timerStarted = true;
+
+            LOG.info(String.format("Heartbeat - [started] - %s - 0ms",
+                    this.configuredUnitTestName));
+        } finally {
+            this.lock.unlock();
```

Comment:
inside each travis CI container, tests run serially. Heartbeat is a quick thing so cant we just synchronize heartbeats on this class? after a jitter wait that is.